pkg/kubelet/kubelet_pods.go

```diff
@@ -1615,13 +1628,19 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 		Type:   v1.PodScheduled,
 		Status: v1.ConditionTrue,
 	})
-	// set HostIP and initialize PodIP/PodIPs for host network pods
+	// set HostIP/HostIPs and initialize PodIP/PodIPs for host network pods
 	if kl.kubeClient != nil {
 		hostIPs, err := kl.getHostIPsAnyWay()
 		if err != nil {
 			klog.V(4).InfoS("Cannot get host IPs", "err", err)
 		} else {
 			s.HostIP = hostIPs[0].String()
+			if utilfeature.DefaultFeatureGate.Enabled(features.PodHostIPs) {
```

Comment:
same

Reply:
Here are the fill IPs, if there is no feature gate, the HostIPs will always be set.